pkg/ottl/functions.go

```diff
@@ -58,18 +58,19 @@ func (p *Parser[K]) buildArgs(inv invocation, fType reflect.Type) ([]reflect.Val
 	for i := 0; i < fType.NumIn(); i++ {
 		argType := fType.In(i)
 
-		switch argType.Kind() {
-		case reflect.Slice:
-			err := p.buildSliceArg(inv, argType, i, &args)
+		switch {
+		case argType.Kind() == reflect.Slice:
+			arg, err := p.buildSliceArg(inv, argType, i)
 			if err != nil {
 				return nil, err
 			}
-			// Slice arguments must be the final argument in an invocation.
-			return args, nil
+			args = append(args, arg)
+			DSLArgumentIndex++
```

Comment:
Can this increment be handled in one place at the end of the for loop outside the switch statement?

Reply:
Good call. Done.